### PR TITLE
docs: improve sse

### LINKED ISSUE #3011: Docs: Document SSE
### Summary

The SSE documentation is currently lacking:

- Docs for `ServerSentEventMessage`
- Sending messages a dicts

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/3011">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/3011/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by euri10**
## Description

- fixes #3011

### CODE REVIEW

**Review by Alc-Alc — APPROVED** [review]

File: docs/examples/responses/sse_responses.py [hunk 1]
```diff
@@ -2,15 +2,27 @@
 from typing import AsyncGenerator
 
 from litestar import Litestar, get
-from litestar.response import ServerSentEvent
+from litestar.response import ServerSentEvent, ServerSentEventMessage
 
 
-async def my_generator() -> AsyncGenerator[bytes, None]:
+async def my_generator() -> AsyncGenerator[ServerSentEventMessage, None]:
```
Comment (peterschutt) [inline docs/examples/responses/sse_responses.py:8]:
```suggestion
async def my_generator() -> AsyncGenerator[Any, None]:
```
The generator is yielding many different types in the example, so perhaps this makes sense?

**Review by peterschutt — APPROVED** [review]
Thanks @euri10!

File: docs/examples/responses/sse_responses.py (hunk 1 above)
Comment (Alc-Alc) [reply docs/examples/responses/sse_responses.py]: what about `ServerSentEvent`?

Comment (peterschutt) [reply docs/examples/responses/sse_responses.py]: Is that a public type?

Comment (Alc-Alc) [reply docs/examples/responses/sse_responses.py]: don't see `_`s so yes?

Comment (euri10) [reply docs/examples/responses/sse_responses.py]: yep, forgot about this one thanks

Comment (peterschutt) [reply docs/examples/responses/sse_responses.py]: I don't see it in the api docs though 🤷‍♂️

Comment (euri10) [reply docs/examples/responses/sse_responses.py]: it's weird, it's not in the docs indeed but in https://github.com/litestar-org/litestar/blob/b9e501fba6ff2ab06ab3afb53d9071a0a8699c55/litestar/types/helper_types.py#L28

is there an issue with the api docs generation ?

Comment (Alc-Alc) [reply docs/examples/responses/sse_responses.py]: I guess this has to be manually added (not automatic) in https://github.com/litestar-org/litestar/blob/b9e501fba6ff2ab06ab3afb53d9071a0a8699c55/docs/reference/types.rst?

Comment (peterschutt) [reply docs/examples/responses/sse_responses.py]: Ref docs only include names listed in `__all__` - so I assume that's why this is.. and if this name isn't in `__all__` then that would mean it's not a public name.

We can add it though - but for the case of this example, is it really that helpful? A reader isn't going to know what the `SSEData` data type represents anyway.

Maybe we should define that type alias in the example itself, wdyt?

Comment (peterschutt) [reply docs/examples/responses/sse_responses.py]: But it is in `__all__` 🫠

Comment (peterschutt) [reply docs/examples/responses/sse_responses.py]: It seems we explicitly name things for the types namespace, and it's just not explicitly named:

https://github.com/litestar-org/litestar/blob/main/docs/reference/types.rst#L141-L150

Comment (peterschutt) [reply docs/examples/responses/sse_responses.py]: > I guess this has to be manually added (not automatic) in https://github.com/litestar-org/litestar/blob/b9e501fba6ff2ab06ab3afb53d9071a0a8699c55/docs/reference/types.rst?

Oops missed this, yeah, this.

Comment (euri10) [reply docs/examples/responses/sse_responses.py]: ok, so should I add it here or open another issue / pr to do that ? or revert to Any ?
I'm ok with both :)

Comment (Alc-Alc) [reply docs/examples/responses/sse_responses.py]: I would probably do it in a new PR so more eyes are on it :)

**Review by JacobCoffee — APPROVED** [review]

**Pull request merged**